exchanges/OrderBookExchange.js

```diff
@@ -92,48 +74,42 @@ module.exports = class OrderBookExchange {
           error: null,
         }
       }
-    } else {
-      // it's a buy
-      const { asks } = book
+    }
+    if (fee !== 0) {
+      result = applyFeeToResult(result, fee)
+    }
+    return { [this.name]: result }
+
+    /**
+     * Calculate the average and total price if the given amount were to be executed on the orderbook.
+     * @param {*} orders the orders from the side of the book the order would be executed against
+     */
+    function calculatePricing(orders) {
```

Comment:
I initially brought this up in https://github.com/grahamperich/ethereum-dex-prices-service/pull/14, but there was some opportunity to simplify the logic. Caught a bug with the logic in the last PR though, so waited until this PR so it could be tested. :P